docs/reference/search/knn-search.asciidoc

```diff
@@ -122,6 +122,12 @@ shard, then merges them to find the top `k` results. Increasing
 `num_candidates` tends to improve the accuracy of the final `k` results.
 ====
 
+`filter`::
+(Optional, <<query-dsl,Query DSL object>>) Query to filter the documents that
+can match. The kNN search will return the top `k` documents that also match
+this filter. The value can be a single query or a list of queries. If `filter`
+is not provided, all documents are allowed to match.
+
```

Comment:
It would be nice to have some json example as with `filter` as well.

Reply:
Good point, I'll actually add an `Examples` section at the end of these docs.